### DISCUSSION TITLE
Maximalwert

### DISCUSSION THREAD

**Discussion by jo3rn** [jo3rn/blockchAIn#11] [Q&A] [answered]
Was passiert bei Aufruf der Methode `mineBlock`, wenn das `nonce` den höchstmöglichen Integer-Wert `Integer.MAX_VALUE` erreicht hat, ohne dass ein passender Hash gefunden wurde?

Anbei die Methode:
```java
public String mineBlock(String difficulty) {
    while (!this.hash.startsWith(difficulty)) {
      nonce++;
      this.hash = calculateHash();
    }
    return this.hash;
  }
```

**Comment by ActionHOSchT** [comment, marked as answer]
![grafik](https://user-images.githubusercontent.com/73703991/113848231-8b21bf80-9798-11eb-94fd-79d5df818e66.png)

![grafik](https://user-images.githubusercontent.com/73703991/113849194-88739a00-9799-11eb-9c3b-972bc4123be7.png)


Integer.MAX_VALUE = 2^31-1 = 2147483647 
Danach wird wieder bei -(2^31) begonnen und erneut durchlaufen. Demnach passiert "nichts".

Block 0 und Block 4 haben den gleichen 'nonce'-Wert (2147483637).

**Reply by ActionHOSchT** [reply]
Hier die Auslastung meines Prozessorkerns:
![grafik](https://user-images.githubusercontent.com/73703991/113860981-f8d4e800-97a6-11eb-92d1-81fdd56aba9d.png)

Weiterer Test mit difficulty = 64* "0".

![grafik](https://user-images.githubusercontent.com/73703991/113861161-2d48a400-97a7-11eb-87c3-ba29645466dc.png)
